### PR TITLE
v1.0.0-rc.8 release

### LINKED ISSUE #6545: Tag Zebra 1.0.0-rc.8 release
## Motivation

We want to tag the next Zebra release.

### PR DESCRIPTION
**Pull request by oxarbitrage**
---
name: Release Checklist Template
about: Checklist of versioning to create a taggable commit for Zebra
title: ''
labels:
assignees: ''

---

### How to Increment Versions

Zebra follows [semantic versioning](https://semver.org). Semantic versions look like: MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]

The [draft `zebrad` changelog](https://github.com/ZcashFoundation/zebra/releases) will have an automatic version bump. This version is based on [the labels on the PRs in the release](https://github.com/ZcashFoundation/zebra/blob/main/.github/release-drafter.yml).

Check that the automatic `zebrad` version increment matches the changes in the release:

<details>

If we're releasing a mainnet network upgrade, it is a `major` release:
1. Increment the `major` version of _*all*_ the Zebra crates.
2. Increment the `patch` version of the tower crates.

If we're not releasing a mainnet network upgrade, check for features, major changes, deprecations, and removals. If this release has any, it is a `minor` release:
1. Increment the `minor` version of `zebrad`.
2. Increment the `pre-release` version of the other crates.
3. Increment the `patch` version of the tower crates.

Otherwise, it is a `patch` release:
1. Increment the `patch` version of `zebrad`.
2. Increment the `pre-release` version of the other crates.
3. Increment the `patch` version of the tower crates.

Zebra's Rust API is not stable or supported, so we keep all the crates on the same beta `pre-release` version.

</details>

### Version Locations

Once you know which versions you want to increment, you can find them in the:

zebrad (rc):
- [x] zebrad `Cargo.toml`
- [x] `README.md`
- [x] `book/src/user/docker.md`

crates (beta):
- [x] zebra-* `Cargo.toml`s

tower (patch):
- [x] tower-* `Cargo.toml`s

auto-generated:
- [x] `Cargo.lock`: run `cargo build` after updating all the `Cargo.toml`s

#### Version Tooling

You can use `fastmod` to interactively find and replace versions.

For example, you can do something like:
```
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-rc.0' '1.0.0-rc.1' zebrad README.md zebra-network/src/constants.rs book/src/user/docker.md
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-beta.15' '1.0.0-beta.16' zebra-*
fastmod --extensions rs,toml,md --fixed-strings '0.2.30' '0.2.31' tower-batch tower-fallback
cargo build
```

If you use `fastmod`, don't update versions in `CHANGELOG.md` or `zebra-dependencies-for-audit.md`.

## README

Update the README to:
- [x] Remove any "Known Issues" that have been fixed
- [x] Update the "Build and Run Instructions" with any new dependencies.
      Check for changes in the `Dockerfile` since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.
- [x] If Zebra has started using newer Rust language features or standard library APIs, update the known working Rust version in the README, book, and `Cargo.toml`s

You can use a command like:
```sh
      fastmod --fixed-strings '1.58' '1.65'
```

## Checkpoints

For performance and security, we want to update the Zebra checkpoints in every release.
You can copy the latest checkpoints from CI by following [the zebra-checkpoints README](https://github.com/ZcashFoundation/zebra/blob/main/zebra-utils/README.md#zebra-checkpoints).

## Missed Dependency Updates

Sometimes `dependabot` misses some dependency updates, or we accidentally turned them off.

Here's how we make sure we got everything:
1. Run `cargo update` on the latest `main` branch, and keep the output
2. If needed, update [deny.toml](https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/continuous-integration.md#fixing-duplicate-dependencies-in-check-denytoml-bans)
3. Open a separate PR with the changes, including the output of `cargo update`

## Change Log

**Important**: Any merge into `main` deletes any edits to the draft changelog.
Once you are ready to tag a release, copy the draft changelog into `CHANGELOG.md`.

We use [the Release Drafter workflow](https://github.com/marketplace/actions/release-drafter) to automatically create a [draft changelog](https://github.com/ZcashFoundation/zebra/releases). We follow the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

To create the final change log:
- [x] Copy the **latest** draft changelog into `CHANGELOG.md` (there can be multiple draft releases)
- [x] Delete any trivial changes. Keep the list of those, to include in the PR
- [x] Combine duplicate changes
- [x] Edit change descriptions so they are consistent, and make sense to non-developers
- [x] Check the category for each change
  - Prefer the "Fix" category if you're not sure

<details>

#### Change Categories

From "Keep a Changelog":
* `Added` for new features.
* `Changed` for changes in existing functionality.
* `Deprecated` for soon-to-be removed features.
* `Removed` for now removed features.
* `Fixed` for any bug fixes.
* `Security` in case of vulnerabilities.

</details>

## Release support constants

Needed for the end of support feature. Please update the following constants [in this file](https://github.com/ZcashFoundation/zebra/blob/main/zebrad/src/components/sync/end_of_support.rs):

- [x] `ESTIMATED_RELEASE_HEIGHT` (required) - Replace with the estimated height you estimate the release will be tagged.
      <details>
      - Find where the Zcash blockchain tip is now by using a [Zcash explorer](https://zcashblockexplorer.com/blocks) or other tool.
      -  Consider there are aprox `1152` blocks per day (with the current Zcash `75` seconds spacing).
      - So for example if you think the release will be tagged somewhere in the next 3 days you can add `1152 * 3` to the current tip height and use that value here.
      </details>

- [x] `EOS_PANIC_AFTER` (optional) - Replace if you want the release to be valid for a different numbers of days into the future. The default here is 120 days.

### Create the Release PR

After you have the version increments, the updated checkpoints, any missed dependency updates,
and the updated changelog:

- [x] Make sure the PRs with the new checkpoint hashes and missed dependencies are already merged
- [x] Push the version increments, the updated changelog and the release constants into a branch
      (for example: `bump-v1.0.0-rc.0` - this needs to be different to the tag name)
- [x] Create a release PR by adding `&template=release-checklist.md` to the comparing url ([Example](https://github.com/ZcashFoundation/zebra/compare/v1.0.0-rc.0-release?expand=1&template=release-checklist.md)).
  - [x] Add the list of deleted changelog entries as a comment to make reviewing easier.
- [x] Freeze the [`batched` queue](https://dashboard.mergify.com/github/ZcashFoundation/repo/zebra/queues) using Mergify.
- [x] Mark all the release PRs as `Critical` priority, so they go in the `urgent` Mergify queue.

### Create the Release

- [x] Once the PR has been merged, create a new release using the draft release as a base, by clicking the Edit icon in the [draft release](https://github.com/ZcashFoundation/zebra/releases)
- [x] Set the tag name to the version tag,
      for example: `v1.0.0-rc.0`
- [x] Set the release to target the `main` branch
- [x] Set the release title to `Zebra ` followed by the version tag,
      for example: `Zebra 1.0.0-rc.0`
- [x] Replace the prepopulated draft changelog in the release description with the final changelog you created;
      starting just _after_ the title `## [Zebra ...` of the current version being released,
      and ending just _before_ the title of the previous release.
- [x] Mark the release as 'pre-release', until it has been built and tested
- [x] Publish the pre-release to GitHub using "Publish Release"
- [x] Delete all the [draft releases from the list of releases](https://github.com/ZcashFoundation/zebra/releases)

## Binary Testing

- [x] Wait until the [Docker binaries have been built on `main`](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-integration-docker.yml), and the quick tests have passed.
      (You can ignore the full sync and `lightwalletd` tests, because they take about a day to run.)
- [x] Wait until the [pre-release deployment machines have successfully launched](https://github.com/ZcashFoundation/zebra/actions/workflows/continous-delivery.yml)
- [x] [Publish the release to GitHub](https://github.com/ZcashFoundation/zebra/releases) by disabling 'pre-release', then clicking "Set as the latest release"
- [x] Wait until [the Docker images have been published](https://github.com/ZcashFoundation/zebra/actions/workflows/release-binaries.yml)
- [x] Test the Docker image using `docker run --tty --interactive zfnd/zebra:1.0.0-rc.<version>`,
      and put the output in a comment on the PR
      
- [x] Un-freeze the [`batched` queue](https://dashboard.mergify.com/github/ZcashFoundation/repo/zebra/queues) using Mergify.

## Telling Zebra Users

- [ ] Post a summary of the important changes in the release in the `#arborist` and `#communications` Slack channels
If the release contains new features (`major` or `minor`), or high-priority bug fixes:
- [ ] Ask the team about doing a blog post

## Release Failures

If building or running fails after tagging:

<details>

1. Fix the bug that caused the failure
2. Increment versions again, following these instructions from the start
3. Update the code and documentation with a **new** git tag
4. Update `CHANGELOG.md` with details about the fix
5. Tag a **new** release

</details>

### CODE REVIEW

**Comment by oxarbitrage** [comment]
### Trivial

- build(deps): bump bitflags from 2.1.0 to 2.2.1 ([#6561](https://github.com/ZcashFoundation/zebra/pull/6561))
- build(deps): bump codecov/codecov-action from 3.1.2 to 3.1.3 ([#6554](https://github.com/ZcashFoundation/zebra/pull/6554))
- build(deps): bump dirs from 5.0.0 to 5.0.1 ([#6592](https://github.com/ZcashFoundation/zebra/pull/6592))
- build(deps): bump displaydoc from 0.2.3 to 0.2.4 ([#6610](https://github.com/ZcashFoundation/zebra/pull/6610))
- build(deps): bump metrics from 0.20.1 to 0.21.0 and metrics-exporter-prometheus from 0.11.0 to 0.12.0 ([#6531](https://github.com/ZcashFoundation/zebra/pull/6531))
- build(deps): bump regex from 1.7.3 to 1.8.1 ([#6555](https://github.com/ZcashFoundation/zebra/pull/6555))
- build(deps): bump reqwest from 0.11.16 to 0.11.17 ([#6591](https://github.com/ZcashFoundation/zebra/pull/6591))
- build(deps): bump sentry from 0.30.0 to 0.31.0 ([#6546](https://github.com/ZcashFoundation/zebra/pull/6546))
- build(deps): bump serde\_with from 2.3.2 to 2.3.3 ([#6588](https://github.com/ZcashFoundation/zebra/pull/6588))
- build(deps): bump serde\_with from 2.3.3 to 3.0.0 ([#6599](https://github.com/ZcashFoundation/zebra/pull/6599))
- build(deps): bump tj-actions/changed-files from 35.8.0 to 35.9.0 ([#6559](https://github.com/ZcashFoundation/zebra/pull/6559))
- build(deps): bump tj-actions/changed-files from 35.9.0 to 35.9.2 ([#6590](https://github.com/ZcashFoundation/zebra/pull/6590))
- build(deps): bump tokio from 1.27.0 to 1.28.0 ([#6572](https://github.com/ZcashFoundation/zebra/pull/6572))
- build(deps): bump tokio-stream from 0.1.12 to 0.1.14 ([#6573](https://github.com/ZcashFoundation/zebra/pull/6573))
- build(deps): bump tokio-util from 0.7.7 to 0.7.8 ([#6570](https://github.com/ZcashFoundation/zebra/pull/6570))
- build(deps): bump tracing-subscriber from 0.3.16 to 0.3.17 ([#6560](https://github.com/ZcashFoundation/zebra/pull/6560))
- build(deps): bump w9jds/firebase-action from 11.24.1 to 11.30.1 ([#6622](https://github.com/ZcashFoundation/zebra/pull/6622))
- change(ci): Generate mainnet checkpoints in CI ([#6550](https://github.com/ZcashFoundation/zebra/pull/6550))
- change(ci): Generate testnet checkpoints in CI ([#6581](https://github.com/ZcashFoundation/zebra/pull/6581))
- change(ci): Split checking for cached state disks into its own workflow ([#6576](https://github.com/ZcashFoundation/zebra/pull/6576))
- change(release): Add arborist and comms summaries to the release checklist ([#6565](https://github.com/ZcashFoundation/zebra/pull/6565))
- change: Add a template for private security issues ([#6594](https://github.com/ZcashFoundation/zebra/pull/6594))
- feat(zebra-network): add user agent argument ([#6601](https://github.com/ZcashFoundation/zebra/pull/6601))
- feat(zebrad): Refuse to run zebrad when release is too old ([#6351](https://github.com/ZcashFoundation/zebra/pull/6351))
- fix(build): sync LWD builder with our `build-docker-image.yml` workflow ([#6634](https://github.com/ZcashFoundation/zebra/pull/6634))
- fix(ci): Avoid inbound service overloads and fix failing tests ([#6537](https://github.com/ZcashFoundation/zebra/pull/6537))
- fix(ci): Delete mainnet and testnet disks separately ([#6582](https://github.com/ZcashFoundation/zebra/pull/6582))
- fix(ci): Update cached disk patch job name ([#6615](https://github.com/ZcashFoundation/zebra/pull/6615))
- fix(log): Fix confusing comments and error messages in zebra-checkpoints ([#6579](https://github.com/ZcashFoundation/zebra/pull/6579))
- fix: Avoid wrong test results for modified JoinSplits ([#6604](https://github.com/ZcashFoundation/zebra/pull/6604))

**Comment by oxarbitrage** [comment]
All these are repeated or not important to mention:

- change(release): Add arborist and comms summaries to the release checklist ([#6565](https://github.com/ZcashFoundation/zebra/pull/6565))
- change(utils): Add a direct connection mode to zebra-checkpoints ([#6516](https://github.com/ZcashFoundation/zebra/pull/6516))

- build(deps): bump tj-actions/changed-files from 35.8.0 to 35.9.0 ([#6559](https://github.com/ZcashFoundation/zebra/pull/6559))
- build(deps): bump tj-actions/changed-files from 35.9.0 to 35.9.2 ([#6590](https://github.com/ZcashFoundation/zebra/pull/6590))
- change(ci): Generate mainnet checkpoints in CI ([#6550](https://github.com/ZcashFoundation/zebra/pull/6550))
- change(ci): Generate testnet checkpoints in CI ([#6581](https://github.com/ZcashFoundation/zebra/pull/6581))
- change(ci): Split checking for cached state disks into its own workflow ([#6576](https://github.com/ZcashFoundation/zebra/pull/6576))
- change(release): Add arborist and comms summaries to the release checklist ([#6565](https://github.com/ZcashFoundation/zebra/pull/6565))
- change(utils): Add a direct connection mode to zebra-checkpoints ([#6516](https://github.com/ZcashFoundation/zebra/pull/6516))
- change: Add a template for private security issues ([#6594](https://github.com/ZcashFoundation/zebra/pull/6594))

- fix(log): Fix confusing comments and error messages in zebra-checkpoints ([#6579](https://github.com/ZcashFoundation/zebra/pull/6579))
- fix(net): Avoid a rare panic when a connection is dropped ([#6566](https://github.com/ZcashFoundation/zebra/pull/6566))
- fix(net): Avoid some self-connection nonce removal attacks ([#6410](https://github.com/ZcashFoundation/zebra/pull/6410))
- fix(net): Reject nodes using ZClassic ports, and warn if configured with those ports ([#6567](https://github.com/ZcashFoundation/zebra/pull/6567))
- fix(test): Prevent finalized\_equals\_pushed\_genesis prop test from panicking ([#6552](https://github.com/ZcashFoundation/zebra/pull/6552))

**Comment by oxarbitrage** [comment]
This is blocked on https://github.com/ZcashFoundation/zebra/pull/6624

File: zebrad/src/components/sync/end_of_support.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use zebra_chain::{
 use crate::application::release_version;
 
 /// The estimated height that this release started to run.
-pub const ESTIMATED_RELEASE_HEIGHT: u32 = 2_026_000;
+pub const ESTIMATED_RELEASE_HEIGHT: u32 = 2_081_448;
```
Comment (teor2345) [inline zebrad/src/components/sync/end_of_support.rs:16]: Note for reviewers: this date is Wed, May 10, 2023, 06:23:52 PM.

(Which is ok, and doesn't need changes, because it's ok if Zebra release candidates shut down a few days early.)

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,37 @@ All notable changes to Zebra are documented in this file.
...
+### Added
+
+- Add ZIP-317 rules to mempool ([#6556](https://github.com/ZcashFoundation/zebra/pull/6556))
+- Add user agent argument to zebra-network crate ([#6601](https://github.com/ZcashFoundation/zebra/pull/6601))
+- Refuse to run zebrad when release is too old ([#6351](https://github.com/ZcashFoundation/zebra/pull/6351))
+
+### Fixed
+
+- Handle randomness generation and invalid random values as errors in cryptographic code ([#6385](https://github.com/ZcashFoundation/zebra/pull/6385))
+- Fix `listen_addr` default deserialization ([#6575](https://github.com/ZcashFoundation/zebra/pull/6575))
```
Comment (teor2345) [inline CHANGELOG.md]: This is the impact that users will see:
```suggestion
- When configured for testnet, automatically use the correct testnet listener port ([#6575](https://github.com/ZcashFoundation/zebra/pull/6575))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,37 @@ All notable changes to Zebra are documented in this file.
...
+- Avoid wrong test results for modified JoinSplits ([#6604](https://github.com/ZcashFoundation/zebra/pull/6604))
```
Comment (teor2345) [inline CHANGELOG.md]: Nitpick: This might not be relevant to users: it's a test that users don't see.
```suggestion
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,37 @@ All notable changes to Zebra are documented in this file.
...
 and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.0.0-rc.8](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-rc.8) - 2023-05-10
+
+This release adds end of support feature which will make `zebrad` binary to halt after the support time of 16 weeks has passed.
+Also, this release adds the ZIP-317 rules to mempool transactions which should help with the Zcash network spam issue.
+
+### Security
+
+- Add a template for private security issues ([#6594](https://github.com/ZcashFoundation/zebra/pull/6594))
```
Comment (teor2345) [inline CHANGELOG.md]: Nitpick: This isn't relevant to users: it is a ticket template that's only for Zebra developers to use.

```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,37 @@ All notable changes to Zebra are documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org).
 
+## [Zebra 1.0.0-rc.8](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-rc.8) - 2023-05-10
+
+This release adds end of support feature which will make `zebrad` binary to halt after the support time of 16 weeks has passed.
```
Comment (teor2345) [inline CHANGELOG.md]: This could be a surprising change for some users, so we might want to explain that it's the same as what `zcashd` does.

```suggestion
Starting in this release, Zebra has implemented an "end of support" halt. Just like `zcashd`, the `zebrad` binary will stop running 16 weeks after the last release date.
```

**Comment by teor2345** [comment]
> This is blocked on #6624

The missed dependency update is also blocked on PR #6624:

> Run cargo update on the latest main branch, and keep the output
> If needed, update [deny.toml](https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/continuous-integration.md#fixing-duplicate-dependencies-in-check-denytoml-bans)
> Open a separate PR with the changes, including the output of cargo update

Did you want some help doing this? I think it will be easier in a separate PR.

File: Cargo.lock [hunk 6]
```diff
@@ -5025,7 +5025,7 @@ dependencies = [
```
Comment (teor2345) [inline Cargo.lock:4823]: > This is blocked on #6624

The missed dependency update is also blocked on PR #6624:

> 1. Run cargo update on the latest main branch, and keep the output
> 2. If needed, update [deny.toml](https://github.com/ZcashFoundation/zebra/blob/main/book/src/dev/continuous-integration.md#fixing-duplicate-dependencies-in-check-denytoml-bans)
> 3. Open a separate PR with the changes, including the output of cargo update

Did you want some help doing this? I think it will be easier in a separate PR.

**Review by teor2345 — COMMENTED** [review]
Looks good, seems like the missed dependency update got skipped.

File: Cargo.lock (hunk 6 above)
Comment (oxarbitrage) [reply Cargo.lock:4823]: Ok, yea. sorry about that and thanks.

Comment (teor2345) [reply Cargo.lock:4823]: This has been merged into this PR, the exact changes are in PR #6656

**Review by teor2345 — APPROVED** [review]

**Comment by teor2345** [comment]
There was an internal GitHub error downloading an action in the testnet test, but testnet tests aren't required to merge, because testnet is unstable anyway:
https://github.com/ZcashFoundation/zebra/actions/runs/4942789784/jobs/8849583735#step:1:25
